DataProcessing/BybitFundingRateDownloader.cs

```diff
@@ -0,0 +1,333 @@
+            var data = url.DownloadData();
+
+            lock (result)
+            {
+                try
+                {
+                    result.AddRange(JsonConvert
+                        .DeserializeObject<ByBitResponse<BybitListResult<BybitFundingRate>>>(data, SerializerSettings)
+                        .Result.List);
+                }
+                catch (Exception)
+                {
+                    Log.Error($"GetData(): deserialization failed {data}");
+                    throw;
+                }
+            }
+        });
+        return result;
+    }
+
+
+    private IEnumerable<DateTime> GetProcessingDates()
+    {
+        if (_deploymentDate.HasValue)
+        {
+            return new[] { _deploymentDate.Value };
+        }
+        else
+        {
+            // everything
+            return Time.EachDay(new DateTime(2019, 9, 13), DateTime.UtcNow.Date);
+        }
+    }
+
+    /// <summary>
+    /// Saves contents to disk, deleting existing zip files
+    /// </summary>
+    /// <param name="destinationFolder">Final destination of the data</param>
+    /// <param name="name">file name</param>
+    /// <param name="contents">Contents to write</param>
+    private void SaveContentToFile(string destinationFolder, string name, Dictionary<DateTime, decimal> contents)
+    {
+        name = name.ToLowerInvariant();
+        var finalPath = Path.Combine(destinationFolder, $"{name}.csv");
+        var existingPath = Path.Combine(_existingInDataFolder, $"{name}.csv");
+
+        if (File.Exists(existingPath))
+        {
+            foreach (var line in File.ReadAllLines(existingPath))
+            {
+                if (string.IsNullOrEmpty(line))
+                {
+                    continue;
+                }
+
+                var parts = line.Split(',');
+                if (parts.Length == 1)
+                {
+                    continue;
+                }
+
+                var time = DateTime.ParseExact(parts[0], "yyyyMMdd HH:mm:ss", CultureInfo.InvariantCulture,
+                    DateTimeStyles.None);
+                var rate = decimal.Parse(parts[1], NumberStyles.Any, CultureInfo.InvariantCulture);
+                if (!contents.ContainsKey(time))
+                {
+                    // use existing unless we have a new value
+                    contents[time] = rate;
+                }
+            }
+        }
+
+        var finalLines = contents.OrderBy(x => x.Key)
+            .Select(x => $"{x.Key:yyyyMMdd HH:mm:ss},{x.Value.ToStringInvariant()}").ToList();
+
+        var tempPath = Path.Combine(Path.GetTempPath(), $"{Guid.NewGuid()}.tmp");
+        File.WriteAllLines(tempPath, finalLines);
+        var tempFilePath = new FileInfo(tempPath);
+        tempFilePath.MoveTo(finalPath, true);
+    }
+
+    private IEnumerable<BybitInstrumentInfo> GetExchangeInfo()
+    {
+        var url = $"{BybitApiEndpoint}/v5/market/instruments-info?limit=1000&status=Trading";
```

Comment:
> limit=1000

Think we should handle this limit being hit, else we would silently start skipping symbols when the limit is hit

Reply:
This is very unlikely to happen when we're fetching the data for each day separately. All the pairs currently have a funding interval of 8 hours meaning we have 3 entries per response. And I don't think they add a coin with a funding interval of 1 minute.

Reply:
This call & endpoint is to get the instruments to fetch the funding rate for though, not the funding rate itself I mean